### ISSUE TITLE
Could not load secring: open /root/.gnupg/secring.gpg: no such file or directory

### ISSUE THREAD

**Issue by zedtux** [jkroepke/helm-secrets#452]
### Current Behavior

When I try to edit an encrypted YAML file on my new Mac, it fails with the following error:

```
❯ helm secrets edit releases/values/dev/gitlab.secrets.yaml
releases/values/dev/gitlab.secrets.yaml
Failed to get the data key required to decrypt the SOPS file.

Group 0: FAILED
  75F85D5C9B18AFCA08ADC6D539D1C80E288155BA: FAILED
    - | could not decrypt data key with PGP key:
      | github.com/ProtonMail/go-crypto/openpgp error: Could not
      | load secring: open /root/.gnupg/secring.gpg: no such file or
      | directory; GPG binary error: exit status 2

Recovery failed because no master key was able to decrypt the file. In
order for SOPS to recover the file, at least one key has to be successful,
but none were.
Error: plugin "secrets" exited with error
```

The private key `75F85D5C9B18AFCA08ADC6D539D1C80E288155BA` is imported and trusted:

```
❯ gpg --list-keys
[keyboxd]
---------
pub   rsa3072 2021-04-28 [SC]
      75F85D5C9B18AFCA08ADC6D539D1C80E288155BA
uid           [ultimate] MyCorp SA <infra@mycorp.co>
sub   rsa3072 2021-04-28 [E]
```

And I can decrypt the file using the `sops` command directly:

```
❯ sops -d releases/values/dev/gitlab.secrets.yaml                                                                                            ─╯
runnerRegistrationToken: [REDACTED]941ghpnSPFBx2yioC_[REDACTED]
```

I don't understand why `github.com/ProtonMail/go-crypto/openpgp` is trying to access `/root/.gnupg/secring.gpg` while the GPG home directory is `/Users/guillaumehain/.gnupg`:

```
❯ gpg --help
gpg (GnuPG) 2.4.3
libgcrypt 1.10.3
Copyright (C) 2023 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /Users/guillaumehain/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2

...
```

### Expected Behavior

When I `edit` a secret file, it succeed at decrypting, updating and saving the file.

### Environment

- Helm Version: 3.12.3
- Helm Secrets Version: 4.1.1
- ArgoCD Version
- OS: macOS 14.5
- Shell: zsh 5.9

### Anything else?

I tried `helm plugin update secrets` which replies :
> Updated plugin: secrets

but I keep getting version 4.1.1 of the plugin.

**Comment by jkroepke** [comment]
Which version of sops do you have installed?

In the lastest version of sops the environment variable XDG_CONFIG_HOME is preferred over the users home.

> I tried helm plugin update secrets which replies :

I feel the integrated helm plugin manager is kind of buggy, since it works with cached git repositories on disk. Try helm plugin remove and du a fresh install.

**Comment by jkroepke** [comment]
> And I can decrypt the file using the sops command directly

Does is also work, if you enter `/bin/sh` and they re-run the command? Does it work out of zsh?

**Comment by zedtux** [comment]
Well... I haven't used helm since months, and therefore I have forgotten the way I have installed it...

As I don't want to mess-up my system, I'm using Docker, so I have created a script file, named `helm` which runs Docker under the hood, so that's why plugin update fails (well doesn't persist I meant), and why the home directory is different.

I'm closing the issue since it is quite sure all my issues are coming from there. Sorry!
